app/src/main/java/com/xmartlabs/scasas/doapp/ui/tasklist/TasksListFragment.java

```diff
@@ -0,0 +1,243 @@
+    newTaskView.setVisibility(View.VISIBLE);
+  }
+
+  @OnClick(R.id.close)
+  void onClickedCloseImageView() {
+    closeNewTaskView();
+    hideKeyboard();
+  }
+
+  @OnClick(R.id.add_task)
+  void onClickedAddTextView() {
+    Task task = Task.builder()
+        .title(titleView.getText().toString().trim())
+        .description(descriptionView.getText().toString().trim())
+        .date(LocalDate.now())
+        .isFinished(false)
+        .user(user)
+        .group(group)
+        .build();
+    insertTask(task);
+    hideKeyboard();
+    closeNewTaskView();
+  }
+
+  private void setFieldsEmpty() {
+    titleView.setText(null);
+    descriptionView.setText(null);
+  }
+
+  private void closeNewTaskView() {
+    fabButtonView.setVisibility(View.VISIBLE);
+    newTaskView.setVisibility(View.GONE);
+    setFieldsEmpty();
+  }
+
+  private void onTappedTask(Task task) {
+    updateTaskState(task);
+  }
+
+  private void updateTaskState(Task task) {
+    taskController.changeTaskState(task)
+        .subscribe(new SingleSubscriber<Task>() {
+          @Override
+          public void onSuccess(Task task) {
+            int message = task.isFinished() ? R.string.good_job : R.string.cmon_you_can_do_this;
+            Toast.makeText(getContext(), message, Toast.LENGTH_SHORT).show();
+          }
+
+          @Override
+          public void onError(Throwable error) {
+            Timber.e(error);
+          }
+        });
+  }
+
+  private void insertTask(Task task) {
+    taskController.insertTask(task)
+        .compose(RxLifecycle.<User, FragmentEvent>bindUntilEvent(lifecycle(), FragmentEvent.DESTROY_VIEW).forSingle())
+        .subscribe(new SingleSubscriber<Task>() {
+          @Override
+          public void onSuccess(Task task) {
+            adapter.addTask(task);
+            noTaskTextView.setVisibility(View.GONE);
+            Toast.makeText(getContext(), R.string.task_added, Toast.LENGTH_SHORT).show();
+          }
+
+          @Override
+          public void onError(Throwable error) {
+            Timber.e(error);
+            Toast.makeText(getContext(), R.string.failed_add_task, Toast.LENGTH_SHORT).show();
+          }
+        });
+  }
+
+  private void getTasks() {
+    taskController.getTasks(user, group)
+        .compose(RxLifecycle.<User, FragmentEvent>bindUntilEvent(lifecycle(), FragmentEvent.DESTROY_VIEW).forSingle())
+        .subscribe(new SingleSubscriber<List<Task>>() {
+          @Override
+          public void onSuccess(List<Task> tasks) {
+            if (!tasks.isEmpty()) {
+              adapter.setTasks(tasks);
+              noTaskTextView.setVisibility(tasks.size() == 0 ? View.VISIBLE : View.GONE);
+            }
+          }
+
+          @Override
+          public void onError(Throwable error) {
+            Timber.e(error);
+          }
+        });
+  }
+
+  private void setupHeader() {
+    if (Objects.equals(group.getName().toUpperCase(), GroupType.SHOP.toString())) {
```

Comment:
GroupType should have an static method, which give you the `GroupType` form a `String`. It will be more clear